### PR TITLE
Updating samples and tests targets and dependencies

### PR DESCRIPTION
**Pull request by fabiocav**
This PR aligns with Azure SDK changes and updates all test and sample application TFMs to the latest supported (net7.0)

All applicable test and sample dependency updates are also applied.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: test/E2ETests/E2EApps/E2EApp/E2EApp.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
     
   <PropertyGroup Condition="$(FunctionsRuntimeVersion) == '3'">
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
```
Comment (kshyju) [inline test/E2ETests/E2EApps/E2EApp/E2EApp.csproj:9]: Oh this is still v3. Should we switch to v4?

**Review by kshyju — APPROVED** [review]

File: test/E2ETests/E2EApps/E2EApp/E2EApp.csproj (hunk 1 above)
Comment (fabiocav) [reply test/E2ETests/E2EApps/E2EApp/E2EApp.csproj:9]: Yes. I'll update this

File: sdk/Sdk/Sdk.csproj [hunk 2]
```diff
@@ -38,6 +38,7 @@
 
     <!-- Marks all packages as 'local only' so they don't end up in the nuspec. -->
     <PackageReference Update="@(PackageReference)" PrivateAssets="All" />
+    <PackageReference Update="SharpZipLib" Version="1.4.1" />
```
Comment (jviau) [inline sdk/Sdk/Sdk.csproj:41]: Can just update line 35 instead.

(Was this VS that inserted the _Update_ item? If so, I may have had a hand in writing that years ago 🤣 )

File: src/ci.yml [hunk 3]
```diff
@@ -14,7 +14,9 @@ pr:
   paths:
     include:
     - src/
-
+    - test/DotNetWorkerTests
```
Comment (jviau) [inline src/ci.yml:17]: We may want to be broader than this even. Changes to props or targets, or other test files could cause CI issues which won't be discovered until a later PR.

**Review by jviau — APPROVED** [review]

File: src/ci.yml (hunk 3 above)
Comment (fabiocav) [reply src/ci.yml:18]: SDK has been added by a separate change, so that is now included.

File: sdk/Sdk/Sdk.csproj (hunk 2 above)
Comment (fabiocav) [reply sdk/Sdk/Sdk.csproj:41]: I'm pretty sure this was indeed done by VS. :)

**Pull request merged**